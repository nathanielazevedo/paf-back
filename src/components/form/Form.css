/** @format */

.form-container {
  position: absolute;
  background-color:rgba(0, 0, 0, 0.3);
  z-index: 998;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 100vw;
  height: 95vh;
}

.form-inner {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: black;
}

.form-title {
  font-size: 2rem;
  color: black;
}

.form-input-container {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  position: relative;
}

.form-input-group {
  display: flex;
  flex-direction: column;
  width: 80%;
  color: transparent;
}

.form-input {
  width: 100%;
  border: none;
  padding: 10px;
  border-radius: 5px;
  background-color: white;
}

.form-button {
  background-color: black;
  padding: 10px;
  color: white;
  border: none;
  border-radius: 15px;
  transition: 0.5s;
  width: 80%;
  font-size: 1.5rem;
  text-transform: uppercase;
  margin-top: 30px;
}


.close {
  top: 0px;
  position: absolute;
  right: 2px;
  border: none;
  font-size: 50px;
  transition: 1s;
}

.show{
  display: block;
}

.hide{
  display: none;
}