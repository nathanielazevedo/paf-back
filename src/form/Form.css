/** @format */

.form-container {
  width: 100vw;
  height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  z-index: 998;
  background: url("../assets/back2.png") center center/cover no-repeat;
  position: absolute;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  text-align: center;
  top: 65px;
}

.form-inner {
  width: 80vw;
  height: fit-content;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  position: relative;
  margin-top: 10%;
}

.form-title {
  font-size: 2rem;
  color: white;
}

.form-input-container {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  position: relative;
}

.form-input-group {
  display: flex;
  flex-direction: column;
  width: 80%;
  color: transparent;
}

.form-input {
  width: 100%;
  border: none;
  padding: 10px;
  border-radius: 5px;
  background-color: white;
}

.form-button {
  background-color: limegreen;
  padding: 10px;
  color: white;
  border: none;
  border-radius: 15px;
  transition: 0.5s;
  width: 80%;
  font-size: 1.5rem;
  text-transform: uppercase;
  margin-top: 30px;
}

.form-button:hover {
  width: 82%;
}

.close {
  top: 0px;
  position: absolute;
  right: 2px;
  border: none;
  font-size: 50px;
  transition: 1s;
}

.close:hover {
  font-size: 35px;
}
