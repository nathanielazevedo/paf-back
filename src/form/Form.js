/** @format */

import "./Form.css";
import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Alert } from "reactstrap";

function MyForm({
  title,
  inputs,
  func,
  after,
  close = null,
  id = null,
  pres = null,
}) {
  const history = useHistory();

  const INITIAL_STATE = {};

  if (pres) {
    inputs.map((i, index) => {
      return (INITIAL_STATE[i] = pres[index]);
    });
  } else {
    inputs.map((i) => {
      return (INITIAL_STATE[i] = "");
    });
  }

  const [formData, setFormData] = useState(INITIAL_STATE);
  const [formErrors, setFormErrors] = useState([]);

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  let closer = close ? (
    <i className="fas fa-times close" onClick={close} data-testid="close"></i>
  ) : null;

  //Submit form to prop func. Reroute to homepage or show errors.
  const handleSubmit = async (evt) => {
    evt.preventDefault();
    let res;
    if (id) {
      res = await func(id, formData);
    } else {
      res = await func(formData);
    }
    if (res.success) {
      setFormData(INITIAL_STATE);
      if (closer) {
        close();
      }
      if (after) {
        history.push(after);
      }
    } else {
      setFormErrors(res.errors);
    }
  };

  return (
    <div className="form-container">
      <div className="form-inner">
        {closer}
        <h3 className="form-title">{title}</h3>
        <form className="form-input-container">
          {inputs.map((m, i) => {
            return (
              <div className="form-input-group" key={i}>
                <label htmlFor={m}>{m}</label>
                <input
                  type="text"
                  name={m}
                  className="form-input"
                  value={formData[m]}
                  onChange={handleChange}
                  placeholder={m}
                  required
                  data-testid={m}
                ></input>
              </div>
            );
          })}
          {formErrors.length
            ? formErrors.map((e, i) => {
                return (
                  <Alert color="danger" key={i}>
                    {" "}
                    {e}
                  </Alert>
                );
              })
            : null}
          <button
            data-testid="button"
            type="submit"
            className="form-button"
            onClick={handleSubmit}
          >
            {title}
          </button>
        </form>
      </div>
    </div>
  );
}

export default MyForm;
