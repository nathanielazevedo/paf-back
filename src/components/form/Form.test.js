/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Form from "./Form";
import { UserProvider } from "../../testUtils";

it("renders without crashing", function () {
  let title = 'Login'
  let inputs = ['Login'];
  let func = 'Login'
  render(
    <MemoryRouter>
      <UserProvider>
        <Form title={title} inputs={inputs} func={func}/>
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
    let title = "Login";
    let inputs = ["Login"];
    let func = "Login";
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Form title={title} inputs={inputs} func={func} />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("displays expected text", function () {
  let title = "Login";
  let inputs = ["Login"];
  let func = "Login";
  const { getAllByText } = render(
    <MemoryRouter>
      <UserProvider>
        <Form title={title} inputs={inputs} func={func} />
      </UserProvider>
    </MemoryRouter>
  );

  expect(getAllByText("Login")[0]).toBeInTheDocument();
});