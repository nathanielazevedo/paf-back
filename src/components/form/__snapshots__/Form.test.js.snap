// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-container"
  >
    <div
      class="form-inner"
    >
      <h3
        class="form-title"
      >
        Login
      </h3>
      <form
        class="form-input-container"
      >
        <div
          class="form-input-group"
        >
          <label
            for="Login"
          >
            Login
          </label>
          <input
            class="form-input"
            data-testid="Login"
            name="Login"
            placeholder="Login"
            required=""
            type="text"
            value=""
          />
        </div>
        <button
          class="form-button"
          data-testid="button"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
